From b2cad1a6f2a0b4ffee4e1ec1255194ed089310fb Mon Sep 17 00:00:00 2001
From: Liang Li <Liang.Li@windriver.com>
Date: Sat, 23 Aug 2008 21:25:05 +0800
Subject: [PATCH] Add fcc-enet related definitions to DTS for sbc8560

1. Add mdio definition in CPM section.
2. Modify the definition of phy2 and phy3, based on sbc8560 board spec.

Signed-off-by: Liang Li <Liang.Li@windriver.com>
---
 arch/powerpc/boot/dts/sbc8560.dts |   30 ++++++++++++++++++------------
 1 files changed, 18 insertions(+), 12 deletions(-)

diff --git a/arch/powerpc/boot/dts/sbc8560.dts b/arch/powerpc/boot/dts/sbc8560.dts
index db3632e..615ff6f 100644
--- a/arch/powerpc/boot/dts/sbc8560.dts
+++ b/arch/powerpc/boot/dts/sbc8560.dts
@@ -156,18 +156,6 @@
 				reg = <0x1a>;
 				device_type = "ethernet-phy";
 			};
-			phy2: ethernet-phy@1b {
-				interrupt-parent = <&mpic>;
-				interrupts = <0x8 0x1>;
-				reg = <0x1b>;
-				device_type = "ethernet-phy";
-			};
-			phy3: ethernet-phy@1c {
-				interrupt-parent = <&mpic>;
-				interrupts = <0x8 0x1>;
-				reg = <0x1c>;
-				device_type = "ethernet-phy";
-			};
 		};
 
 		enet0: ethernet@24000 {
@@ -239,6 +227,24 @@
 				compatible = "fsl,mpc8560-cpm-pic", "fsl,cpm2-pic";
 			};
 
+			mdio@90d40 {				/* For FCCs */
+				#address-cells = <1>;
+				#size-cells = <0>;
+				compatible = "fsl,cpm2-mdio-bitbang";
+				reg = <0x90d00 0x14>;
+				fsl,mdio-pin = <9>;
+				fsl,mdc-pin = <10>;
+
+				phy2: ethernet-phy@0 {
+					reg = <0x0>;
+					device_type = "ethernet-phy";
+				};
+				phy3: ethernet-phy@1 {
+					reg = <0x1>;
+					device_type = "ethernet-phy";
+				};
+			};
+
 			enet2: ethernet@91320 {
 				device_type = "network";
 				compatible = "fsl,mpc8560-fcc-enet",
-- 
1.5.5.1

